From 20b51b686b76b747f09f2c4bca1c63cf76aee81d Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 3 Mar 2015 12:31:48 +0100
Subject: [PATCH] network/netroot: setup interface correctly, before bailing
 out

at least setup the interface, even if no netroot has to be done
---
 modules.d/40network/netroot | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/modules.d/40network/netroot b/modules.d/40network/netroot
index 8a2f852..cde039e 100755
--- a/modules.d/40network/netroot
+++ b/modules.d/40network/netroot
@@ -61,17 +61,6 @@ else
     rootok="1"
 fi
 
-# Check: do we really know how to handle (net)root?
-[ -z "$root" ] && die "No or empty root= argument"
-[ -z "$rootok" ] && die "Don't know how to handle 'root=$root'"
-
-handler=${netroot%%:*}
-handler=${handler%%4}
-handler="/sbin/${handler}root"
-if [ -z "$netroot" ] || [ ! -e "$handler" ] ; then
-    die "No handler for netroot type '$netroot'"
-fi
-
 # We're here, so we can assume that upping interfaces is now ok
 [ -z "$IFACES" ] && IFACES="$netif"
 
@@ -100,6 +89,17 @@ if [ -z "$dest" ] ; then
 fi
 [ -n "$dest" ] && arping -q -f -w 60 -I $netif $dest
 
+# Check: do we really know how to handle (net)root?
+[ -z "$root" ] && die "No or empty root= argument"
+[ -z "$rootok" ] && die "Don't know how to handle 'root=$root'"
+
+handler=${netroot%%:*}
+handler=${handler%%4}
+handler="/sbin/${handler}root"
+if [ -z "$netroot" ] || [ ! -e "$handler" ] ; then
+    die "No handler for netroot type '$netroot'"
+fi
+
 # Source netroot hooks before we start the handler
 source_all netroot
 
